File: app/src/screens/BulletinDetailScreen.js
import React, { useState } from 'react';
import { View, Text, FlatList, ScrollView, TouchableOpacity, StyleSheet } from 'react-native';
import { colors, spacing, radius } from '../theme';
import { Tabs, EmptyState } from '../ui';
import { useBulletinDetail } from '../hooks/useBulletinHistory';
import MatchPredictionRow from '../components/MatchPredictionRow';
import ResultComparisonCard from '../components/ResultComparisonCard';
import DashboardChartCard from '../components/DashboardChartCard';
import LoadingState from '../components/LoadingState';
import ErrorState from '../components/ErrorState';
import DemoDataBanner from '../components/DemoDataBanner';
import { BULLETIN_STATUS_LABEL } from '../types/bulletin';
import { ERROR_TAG_LABEL } from '../types/analysis';
import { matchDate } from '../utils';

const TAB_NAMES = ['Maçlar', 'Kilitli Analiz', 'Kuponum', 'Sonuçlar', 'Sistem Karnesi'];

// B) Bülten Detay Ekranı — sekmeli: Maçlar / Kilitli Analiz / Kuponum /
// Sonuçlar / Sistem Karnesi. Hem aktif hem geçmiş bültenler için kullanılır.
export default function BulletinDetailScreen({ route, navigation }) {
  const { bulletinId } = route.params || {};
  const { bulletin, snapshot, coupon, loading, error, reload } = useBulletinDetail(bulletinId);
  const [tab, setTab] = useState(TAB_NAMES[0]);

  if (loading && !bulletin) return <LoadingState message="Bülten detayı yükleniyor…" />;
  if (error) return <ErrorState message={error} onRetry={reload} />;
  if (!bulletin) return null;

  const sealed = !!(snapshot && snapshot.isLocked);
  const shortHash = snapshot?.shortHash || null;
  const rs = bulletin.resultSummary;

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.title}>Bülten {bulletin.bulletinNo}</Text>
        <Text style={styles.muted}>
          {BULLETIN_STATUS_LABEL[bulletin.status]}
          {bulletin.lockedAt ? ` · Kilit: ${matchDate(bulletin.lockedAt).day} ${matchDate(bulletin.lockedAt).time}` : ''}
          {bulletin.status === 'completed' && rs ? ` · Sistem: ${rs.systemCorrect}/${rs.totalCount}` : ''}
        </Text>
        {sealed ? (
          <Text style={styles.sealLine}>
            🔏 Mühürlü Analiz — bu bültenin tahmin ve analizleri değiştirilemez
            {shortHash ? ` · Doğrulama: #${shortHash}` : ''}
          </Text>
        ) : bulletin.freezeAt ? (
          <Text style={styles.freezeLine}>
            🔒 Analizler {matchDate(bulletin.freezeAt).day} {matchDate(bulletin.freezeAt).time} itibarıyla kilitlenecek
          </Text>
        ) : null}
      </View>

      {bulletin._demo && (
        <DemoDataBanner note="Arşiv sunucusuna ulaşılamadı — bu bülten detayı ÖRNEK veridir, gerçek Spor Toto bülteni/sonucu değildir." />
      )}

      <Tabs tabs={TAB_NAMES} active={tab} onChange={setTab} />

      <View style={{ flex: 1 }}>
        {tab === 'Maçlar' && <MatchesTab bulletin={bulletin} />}
        {tab === 'Kilitli Analiz' && <LockedAnalysisTab bulletin={bulletin} snapshot={snapshot} />}
        {tab === 'Kuponum' && <KuponTab bulletin={bulletin} coupon={coupon} navigation={navigation} />}
        {tab === 'Sonuçlar' && <ResultsTab bulletin={bulletin} snapshot={snapshot} />}
        {tab === 'Sistem Karnesi' && <SystemKarnesiTab snapshot={snapshot} />}
      </View>
    </View>
  );
}

function MatchesTab({ bulletin }) {
  return (
    <FlatList
      data={bulletin.matches}
      keyExtractor={(m) => m.id}
      renderItem={({ item }) => <MatchPredictionRow match={item} analysis={null} />}
      contentContainerStyle={styles.listPad}
    />
  );
}

function LockedAnalysisTab({ bulletin, snapshot }) {
  if (!snapshot) {
    const f = bulletin.freezeAt ? matchDate(bulletin.freezeAt) : null;
    return (
      <EmptyState
        icon="🔒"
        title="Analiz henüz mühürlenmedi"
        message={f
          ? `Analizler ${f.day} ${f.time} itibarıyla (ilk maçtan 5 dk önce) kilitlenecek ve arşive mühürlenecek.`
          : 'Bu bülten için henüz bir analiz kaydı oluşmadı.'}
      />
    );
  }
  const lockD = snapshot.lockedAt ? matchDate(snapshot.lockedAt) : null;
  return (
    <FlatList
      data={bulletin.matches}
      keyExtractor={(m) => m.id}
      renderItem={({ item }) => (
        <MatchPredictionRow match={item} analysis={snapshot.matchesAnalysis.find((a) => String(a.matchId) === String(item.id))} />
      )}
      contentContainerStyle={styles.listPad}
      ListHeaderComponent={
        <Text style={styles.lockNote}>
          {snapshot.isLocked
            ? `🔏 Mühürlü Analiz${lockD ? ` · ${lockD.day} ${lockD.time}` : ''} — maçlar başlamadan önce donduruldu; bu bültenin tahmin ve analizleri değiştirilemez. Sonradan gelen skor/istatistik bu kayda işlemez.${snapshot.shortHash ? ` Doğrulama: #${snapshot.shortHash}` : ''}`
            : '🔓 Analiz henüz düzenlenebilir (ilk maç başlamadı).'}
        </Text>
      }
    />
  );
}

function KuponTab() {
  // ESKİ demo kupon akışı KALDIRILDI. Kuponlar artık tek yerden, GERÇEK
  // veriyle çalışan Kupon Merkezi'nden yönetilir (alt bar → Kuponlarım).
  return (
    <View style={styles.listPad}>
      <EmptyState
        icon="🎟️"
        title="Kuponlar Kupon Merkezi'nde"
        message="Kupon oluşturma, sonuç ve paylaşım artık alt bardaki Kuponlarım (Kupon Merkezi) bölümünde — gerçek veriyle çalışır, demo kupon üretilmez."
      />
    </View>
  );
}
function ResultsTab({ bulletin, snapshot }) {
  if (!snapshot) return <EmptyState icon="📭" title="Sonuç yok" message="Bu hafta için analiz/sonuç kaydı yok." />;
  const rows = snapshot.matchesAnalysis.filter((m) => m.resultInfo && m.resultInfo.actualResult);
  if (!rows.length) {
    return <EmptyState icon="⏳" title="Resmî sonuçlar henüz açıklanmadı" message="Resmî 90 dakika sonuçları (1/X/2) geldikçe burada görünecek." />;
  }
  const dataD = rows[0]?.dataTimestamp ? matchDate(rows[0].dataTimestamp) : null;
  return (
    <FlatList
      data={rows}
      keyExtractor={(m) => String(m.matchId)}
      renderItem={({ item }) => {
        const match = bulletin.matches.find((m2) => String(m2.id) === String(item.matchId));
        return (
          <ResultComparisonCard
            orderNo={match?.orderNo ?? item.orderNo}
            homeTeam={match?.homeTeam.name}
            awayTeam={match?.awayTeam.name}
            userPick={null}
            systemPick={item.prediction}
            actualResult={item.resultInfo.actualResult}
            isCorrect={item.resultInfo.systemCorrect}
          />
        );
      }}
      contentContainerStyle={styles.listPad}
      ListHeaderComponent={
        <Text style={styles.lockNote}>
          {`Soldaki tahminler ${dataD ? `${dataD.day} ${dataD.time}` : 'kilit anı'} verisiyle MÜHÜRLENMİŞ hali; sağdaki sonuçlar resmî Spor Toto 90 dk sonucudur. İkisi ayrı kayıtlarda tutulur — sonuç, analizi geriye dönük değiştirmez.`}
        </Text>
      }
    />
  );
}

function SystemKarnesiTab({ snapshot }) {
  if (!snapshot) return <EmptyState icon="📊" title="Veri yok" message="Bu bülten için sistem karnesi oluşmadı." />;
  const resolved = snapshot.matchesAnalysis.filter((m) => m.resultInfo && m.resultInfo.systemCorrect !== null);
  if (!resolved.length) {
    return <EmptyState icon="⏳" title="Henüz sonuç yok" message="Maçlar sonuçlandıkça sistem karnesi burada oluşacak." />;
  }
  const correct = resolved.filter((m) => m.resultInfo.systemCorrect).length;
  const rate = Math.round((correct / resolved.length) * 100);
  const errorCounts = {};
  resolved.forEach((m) => {
    if (!m.resultInfo.systemCorrect && m.resultInfo.errorTag) {
      errorCounts[m.resultInfo.errorTag] = (errorCounts[m.resultInfo.errorTag] || 0) + 1;
    }
  });
  const wrong = resolved.length - correct;
  return (
    <ScrollView contentContainerStyle={styles.listPad}>
      <DashboardChartCard
        title={`Sistem başarısı: ${correct}/${resolved.length} (%${rate})`}
        rows={[{ label: 'Doğru oran', value: rate, color: colors.green }]}
      />
      {Object.keys(errorCounts).length > 0 && (
        <DashboardChartCard
          title="Hata dağılımı"
          rows={Object.entries(errorCounts).map(([tag, n]) => ({
            label: `${ERROR_TAG_LABEL[tag] || tag} (${n})`,
            value: wrong ? Math.round((n / wrong) * 100) : 0,
            color: colors.red,
          }))}
        />
      )}
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: colors.bg },
  header: { padding: spacing.lg, backgroundColor: colors.card, borderBottomWidth: 1, borderBottomColor: colors.border },
  title: { color: colors.text, fontSize: 18, fontWeight: '800' },
  muted: { color: colors.textMuted, fontSize: 12.5, marginTop: 2 },
  sealLine: { color: colors.green, fontSize: 11.5, fontWeight: '800', marginTop: 6, lineHeight: 15 },
  freezeLine: { color: colors.orange, fontSize: 11.5, fontWeight: '800', marginTop: 6, lineHeight: 15 },
  listPad: { padding: spacing.md, paddingBottom: spacing.xl },
  lockNote: { color: colors.textMuted, fontSize: 12, fontStyle: 'italic', marginBottom: spacing.sm, lineHeight: 17 },
  cta: { marginTop: spacing.md, backgroundColor: colors.primary, borderRadius: radius.md, paddingVertical: 13, alignItems: 'center' },
  ctaTxt: { color: colors.bg, fontSize: 14.5, fontWeight: '800' },
  couponHead: { marginBottom: spacing.sm },
  couponHeadTxt: { color: colors.textMuted, fontSize: 12.5, fontWeight: '700', marginBottom: spacing.sm },
  couponBtnRow: { flexDirection: 'row', gap: 8, flexWrap: 'wrap' },
  editBtn: { backgroundColor: colors.cardAlt, borderRadius: radius.sm, paddingHorizontal: 12, paddingVertical: 8 },
  editBtnTxt: { color: colors.primary, fontSize: 12, fontWeight: '800' },
});

File: lib/features/bulletin_history/bulletin_detail_screen.dart
// KAYNAK: app/src/screens/BulletinDetailScreen.js +
//         app/src/hooks/useBulletinHistory.js → useBulletinDetail
// BİREBİR çeviri.
//
// B) Bülten Detay Ekranı — sekmeli: Maçlar / Kilitli Analiz / Kuponum /
// Sonuçlar / Sistem Karnesi. Hem aktif hem geçmiş bültenler için kullanılır.
//
// EKRANIN TAŞIDIĞI KURAL: mühürlü analiz ile resmî sonuç AYRI kayıtlardır.
// Sonuç, analizi geriye dönük DEĞİŞTİRMEZ; ekran ikisini yan yana gösterir.

import 'package:flutter/material.dart';

import '../../core/services/archive_client.dart';
import '../../core/services/bulletin_history_service.dart';
import '../../core/theme/tokens.dart';
import '../../core/types/bulletin.dart';
import '../../core/utils.dart';
import '../../widgets/app_ui.dart';
import '../../widgets/states.dart';
import '../../widgets/tabs.dart';
import 'history_widgets.dart';

const List<String> _tabNames = [
  'Maçlar',
  'Kilitli Analiz',
  'Kuponum',
  'Sonuçlar',
  'Sistem Karnesi',
];

class BulletinDetailScreen extends StatefulWidget {
  const BulletinDetailScreen({super.key, required this.bulletinId});

  final Object bulletinId;

  @override
  State<BulletinDetailScreen> createState() => _BulletinDetailScreenState();
}

class _BulletinDetailScreenState extends State<BulletinDetailScreen> {
  Map<String, dynamic>? _bulletin;
  Map<String, dynamic>? _snapshot;
  bool _loading = true;
  String? _error;
  String _tab = _tabNames[0];

  @override
  void initState() {
    super.initState();
    _reload();
  }

  Future<void> _reload() async {
    setState(() {
      _loading = true;
      _error = null;
    });
    try {
      // Kaynakta Promise.all — ikisi PARALEL istenir. Snapshot yoksa (aktif
      // bülten) null döner ve bu bir hata DEĞİLDİR.
      final sonuclar = await Future.wait([
        getBulletinById(widget.bulletinId),
        getSnapshot(widget.bulletinId),
      ]);
      final bulletin = sonuclar[0];
      if (bulletin == null) throw Exception('Bülten bulunamadı.');
      if (!mounted) return;
      setState(() {
        _bulletin = bulletin;
        _snapshot = sonuclar[1];
        _loading = false;
      });
    } catch (e) {
      if (mounted) {
        setState(() {
          _loading = false;
          _error = humanArchiveError(e);
        });
      }
    }
  }

  @override
  Widget build(BuildContext context) {
    final b = _bulletin;

    if (_loading && b == null) {
      return _kabuk(const LoadingState(message: 'Bülten detayı yükleniyor…'));
    }
    if (_error != null) {
      return _kabuk(
        SingleChildScrollView(
          padding: const EdgeInsets.symmetric(vertical: Spacing.lg),
          child: ErrorState(message: _error, onRetry: _reload),
        ),
      );
    }
    if (b == null) return _kabuk(const SizedBox.shrink());

    final snap = _snapshot;
    final sealed = snap != null && snap['isLocked'] == true;
    final shortHash = snap?['shortHash'];
    final lockedAt = b['lockedAt'];
    final freezeAt = b['freezeAt'];
    final durum = '${b['status']}';

    final altSatir = StringBuffer(kBulletinStatusLabel[durum] ?? durum);
    if (lockedAt != null) {
      final d = matchDate('$lockedAt');
      altSatir.write(' · Kilit: ${d.day} ${d.time}');
    }
    // TEK ÖLÇÜ (2026-08-11): başlıktaki 'Sistem: X/Y' kupon kapsaması
    // sayısıydı — karneyle çelişen yüzdelerin kaynaklarından biri. Kaldırıldı;
    // sistem başarısı yalnız 'Sistem Karnesi' sekmesinde (tekli) yazar.

    return _kabuk(
      Column(
        children: [
          Container(
            width: double.infinity,
            padding: const EdgeInsets.all(Spacing.lg),
            decoration: BoxDecoration(
              color: AppColors.card,
              border: Border(bottom: BorderSide(color: AppColors.border)),
            ),
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  'Bülten ${b['bulletinNo']}',
                  style: TextStyle(
                    color: AppColors.text,
                    fontSize: 18,
                    fontWeight: AppFont.heavy,
                  ),
                ),
                const SizedBox(height: 2),
                Text(
                  altSatir.toString(),
                  style: TextStyle(color: AppColors.textMuted, fontSize: 12.5),
                ),
                if (sealed)
                  Padding(
                    padding: const EdgeInsets.only(top: 6),
                    child: Text(
                      '🔏 Mühürlü Analiz — bu bültenin tahmin ve analizleri değiştirilemez'
                      '${shortHash != null ? ' · Doğrulama: #$shortHash' : ''}',
                      style: const TextStyle(
                        color: AppColors.green,
                        fontSize: 11.5,
                        fontWeight: AppFont.heavy,
                        height: 15 / 11.5,
                      ),
                    ),
                  )
                else if (freezeAt != null)
                  Padding(
                    padding: const EdgeInsets.only(top: 6),
                    child: Text(
                      '🔒 Analizler ${matchDate('$freezeAt').day} ${matchDate('$freezeAt').time} itibarıyla kilitlenecek',
                      style: const TextStyle(
                        color: AppColors.orange,
                        fontSize: 11.5,
                        fontWeight: AppFont.heavy,
                        height: 15 / 11.5,
                      ),
                    ),
                  ),
              ],
            ),
          ),
          if (b['_demo'] == true)
            const DemoDataBanner(
              note:
                  'Arşiv sunucusuna ulaşılamadı — bu bülten detayı ÖRNEK veridir, gerçek Spor Toto bülteni/sonucu değildir.',
            ),
          AppTabs(
            tabs: _tabNames,
            active: _tab,
            onChange: (t) => setState(() => _tab = t),
          ),
          Expanded(child: _icerik(b, snap)),
        ],
      ),
    );
  }

  Widget _kabuk(Widget govde) => Scaffold(
    appBar: AppBar(title: const Text('Bülten Detayı')),
    body: govde,
  );

  Widget _icerik(Map<String, dynamic> b, Map<String, dynamic>? snap) =>
      switch (_tab) {
        'Maçlar' => _maclarSekmesi(b),
        'Kilitli Analiz' => _kilitliAnalizSekmesi(b, snap),
        'Kuponum' => _kuponSekmesi(),
        'Sonuçlar' => _sonuclarSekmesi(b, snap),
        _ => _sistemKarnesiSekmesi(snap),
      };

  static const _listPad = EdgeInsets.fromLTRB(
    Spacing.md,
    Spacing.md,
    Spacing.md,
    Spacing.xl,
  );

  Widget _maclarSekmesi(Map<String, dynamic> b) {
    final matches = (b['matches'] as List?) ?? const [];
    return ListView.builder(
      padding: _listPad,
      itemCount: matches.length,
      itemBuilder: (_, i) =>
          MatchPredictionRow(match: matches[i] as Map, analysis: null),
    );
  }

  Widget _kilitliAnalizSekmesi(
    Map<String, dynamic> b,
    Map<String, dynamic>? snap,
  ) {
    if (snap == null) {
      final freezeAt = b['freezeAt'];
      final f = freezeAt != null ? matchDate('$freezeAt') : null;
      return SingleChildScrollView(
        padding: _listPad,
        child: EmptyState(
          icon: '🔒',
          title: 'Analiz henüz mühürlenmedi',
          message: f != null
              ? 'Analizler ${f.day} ${f.time} itibarıyla (ilk maçtan 5 dk önce) kilitlenecek ve arşive mühürlenecek.'
              : 'Bu bülten için henüz bir analiz kaydı oluşmadı.',
        ),
      );
    }

    final matches = (b['matches'] as List?) ?? const [];
    final analizler = (snap['matchesAnalysis'] as List?) ?? const [];
    final lockedAt = snap['lockedAt'];
    final lockD = lockedAt != null ? matchDate('$lockedAt') : null;
    final shortHash = snap['shortHash'];

    final not = snap['isLocked'] == true
        ? '🔏 Mühürlü Analiz${lockD != null ? ' · ${lockD.day} ${lockD.time}' : ''} '
              '— maçlar başlamadan önce donduruldu; bu bültenin tahmin ve analizleri '
              'değiştirilemez. Sonradan gelen skor/istatistik bu kayda işlemez.'
              '${shortHash != null ? ' Doğrulama: #$shortHash' : ''}'
        : '🔓 Analiz henüz düzenlenebilir (ilk maç başlamadı).';

    return ListView.builder(
      padding: _listPad,
      itemCount: matches.length + 1,
      itemBuilder: (_, i) {
        if (i == 0) {
          return Padding(
            padding: const EdgeInsets.only(bottom: Spacing.sm),
            child: Text(
              not,
              style: TextStyle(
                color: AppColors.textMuted,
                fontSize: 12,
                fontStyle: FontStyle.italic,
                height: 17 / 12,
              ),
            ),
          );
        }
        final m = matches[i - 1] as Map;
        Map? analiz;
        for (final a in analizler) {
          if ('${(a as Map)['matchId']}' == '${m['id']}') {
            analiz = a;
            break;
          }
        }
        return MatchPredictionRow(match: m, analysis: analiz);
      },
    );
  }

  /// ESKİ demo kupon akışı KALDIRILDI. Kuponlar artık tek yerden, GERÇEK
  /// veriyle çalışan Kupon Merkezi'nden yönetilir (alt bar → Kuponlarım).
  Widget _kuponSekmesi() => const SingleChildScrollView(
    padding: _listPad,
    child: EmptyState(
      icon: '🎟️',
      title: "Kuponlar Kupon Merkezi'nde",
      message:
          'Kupon oluşturma, sonuç ve paylaşım artık alt bardaki Kuponlarım (Kupon Merkezi) bölümünde — gerçek veriyle çalışır, demo kupon üretilmez.',
    ),
  );

  Widget _sonuclarSekmesi(Map<String, dynamic> b, Map<String, dynamic>? snap) {
    if (snap == null) {
      return const SingleChildScrollView(
        padding: _listPad,
        child: EmptyState(
          icon: '📭',
          title: 'Sonuç yok',
          message: 'Bu hafta için analiz/sonuç kaydı yok.',
        ),
      );
    }

    final rows = [
      for (final m in (snap['matchesAnalysis'] as List?) ?? const [])
        if ((m as Map)['resultInfo'] is Map &&
            (m['resultInfo'] as Map)['actualResult'] != null)
          m,
    ];

    if (rows.isEmpty) {
      return const SingleChildScrollView(
        padding: _listPad,
        child: EmptyState(
          icon: '⏳',
          title: 'Resmî sonuçlar henüz açıklanmadı',
          message:
              'Resmî 90 dakika sonuçları (1/X/2) geldikçe burada görünecek.',
        ),
      );
    }

    final ts = rows.first['dataTimestamp'];
    final dataD = ts != null ? matchDate('$ts') : null;
    final matches = (b['matches'] as List?) ?? const [];

    return ListView.builder(
      padding: _listPad,
      itemCount: rows.length + 1,
      itemBuilder: (_, i) {
        if (i == 0) {
          return Padding(
            padding: const EdgeInsets.only(bottom: Spacing.sm),
            child: Text(
              'Soldaki tahminler ${dataD != null ? '${dataD.day} ${dataD.time}' : 'kilit anı'} '
              'verisiyle MÜHÜRLENMİŞ hali; sağdaki sonuçlar resmî Spor Toto 90 dk '
              'sonucudur. İkisi ayrı kayıtlarda tutulur — sonuç, analizi geriye '
              'dönük değiştirmez.',
              style: TextStyle(
                color: AppColors.textMuted,
                fontSize: 12,
                fontStyle: FontStyle.italic,
                height: 17 / 12,
              ),
            ),
          );
        }
        final item = rows[i - 1];
        Map? match;
        for (final m2 in matches) {
          if ('${(m2 as Map)['id']}' == '${item['matchId']}') {
            match = m2;
            break;
          }
        }
        final ri = item['resultInfo'] as Map;
        // TEK ÖLÇÜ (2026-08-11): tik TEKLİ ana tahmine göredir (karneyle
        // aynı kural). Ana tahmin mühürde yoksa kupon önerisi tiksiz görünür
        // — "yanlış" denmez, değerlendirmeye girmez.
        return ResultComparisonCard(
          orderNo: match?['orderNo'] ?? item['orderNo'],
          homeTeam: (match?['homeTeam'] as Map?)?['name'] as String?,
          awayTeam: (match?['awayTeam'] as Map?)?['name'] as String?,
          userPick: null,
          systemPick: (item['anaTahmin'] ?? item['prediction']) as String?,
          actualResult: ri['actualResult'] as String?,
          isCorrect: ri['anaTahminCorrect'] as bool?,
        );
      },
    );
  }

  Widget _sistemKarnesiSekmesi(Map<String, dynamic>? snap) {
    if (snap == null) {
      return const SingleChildScrollView(
        padding: _listPad,
        child: EmptyState(
          icon: '📊',
          title: 'Veri yok',
          message: 'Bu bülten için sistem karnesi oluşmadı.',
        ),
      );
    }

    // TEK ÖLÇÜ (2026-08-11): bu sekme TEKLİ mühürlü ana tahmini sayar —
    // Sistem Karnesi ekranıyla AYNI kural, aynı sayı. (Önceden çoklu kupon
    // kapsamasını sayıyordu; karneyle çelişen %93/%36 karmaşasının
    // kaynağıydı.) anaTahminCorrect null ise maç değerlendirilmemiştir;
    // paydaya katılmaz — katsaydı oran yapay düşerdi.
    final resolved = [
      for (final m in (snap['matchesAnalysis'] as List?) ?? const [])
        if ((m as Map)['resultInfo'] is Map &&
            (m['resultInfo'] as Map)['anaTahminCorrect'] != null)
          m,
    ];

    if (resolved.isEmpty) {
      return const SingleChildScrollView(
        padding: _listPad,
        child: EmptyState(
          icon: '⏳',
          title: 'Henüz sonuç yok',
          message: 'Maçlar sonuçlandıkça sistem karnesi burada oluşacak.',
        ),
      );
    }

    final correct = resolved
        .where((m) => (m['resultInfo'] as Map)['anaTahminCorrect'] == true)
        .length;
    final rate = (correct / resolved.length * 100).round();
    final wrong = resolved.length - correct;

    final errorCounts = <String, int>{};
    for (final m in resolved) {
      final ri = m['resultInfo'] as Map;
      final tag = ri['errorTag'];
      if (ri['anaTahminCorrect'] != true && tag != null) {
        errorCounts['$tag'] = (errorCounts['$tag'] ?? 0) + 1;
      }
    }

    return ListView(
      padding: _listPad,
      children: [
        DashboardChartCard(
          title: 'Ana tahmin başarısı: $correct/${resolved.length} (%$rate)',
          rows: [(label: 'Doğru oran', value: rate, color: AppColors.green)],
        ),
        if (errorCounts.isNotEmpty)
          DashboardChartCard(
            title: 'Hata dağılımı',
            rows: [
              for (final e in errorCounts.entries)
                (
                  label: '${kErrorTagLabel[e.key] ?? e.key} (${e.value})',
                  value: wrong > 0 ? (e.value / wrong * 100).round() : 0,
                  color: AppColors.red,
                ),
            ],
          ),
      ],
    );
  }
}
